extensions/amp-ad-network-adsense-impl/0.1/responsive-state.js

```diff
@@ -103,6 +103,12 @@ export class ResponsiveState {
     if (element.hasAttribute('data-auto-format')) {
       return Promise.resolve(null);
     }
+
+    // If the user already has a wide viewport layout, we don't upgrade to responsive.
+    if (!ResponsiveState.isLayoutViewportNarrow_(element)) {
```

Comment:
IIUC we are just pushing this logic into the existing experiment. Is that the desired behavior?

Reply:
My understanding is that maybeUpgradeToResponsive will have a few conditions check before upgrading the ad to a responsive one.

My changes is to add an additional condition to ensure that the users with wide viewport like desktop users will **not** be upgraded to responsive. Inside my [Google doc](https://docs.google.com/document/d/16umi6J2d6hajbdt27skhbzH-7w2tIZ4u922kxVWl1Oo/edit#heading=h.l8jetkktdduy), it has some internal tickets for the broken user experience. Thanks.

Reply:
Ok sgtm.